staging/src/k8s.io/legacy-cloud-providers/gce/gce.go

```diff
@@ -439,6 +439,9 @@ func CreateGCECloud(config *CloudConfig) (*Cloud, error) {
 		service.BasePath = fmt.Sprintf("%sprojects/", config.APIEndpoint)
 		serviceBeta.BasePath = fmt.Sprintf("%sprojects/", strings.Replace(config.APIEndpoint, "v1", "beta", -1))
 		serviceAlpha.BasePath = fmt.Sprintf("%sprojects/", strings.Replace(config.APIEndpoint, "v1", "alpha", -1))
+	} else {
+		klog.Infof("Reset basePath %v with APIEndpoint: %v", service.BasePath, config.APIEndpoint)
+		service.BasePath = fmt.Sprintf("%sprojects/", service.BasePath)
```

Comment:
this area looks surprisingly fragile... I have a hard time following the changes to the unit tests and knowing whether what is being tested reflects reality coming from the API, from the config, from the cloud provider library or as fixed up here...